client/my-sites/email/email-management/home/email-plan-mailboxes-list.jsx

```diff
@@ -13,42 +15,70 @@ import SectionHeader from 'calypso/components/section-header';
 import Badge from 'calypso/components/badge';
 import { isEmailUserAdmin } from 'calypso/lib/emails';
 
-class EmailPlanMailboxesList extends React.Component {
-	render() {
-		const { emails, translate } = this.props;
-
-		if ( ! emails || emails.length < 1 ) {
-			return null;
-		}
-
-		const emailsItems = emails.map( ( email ) => {
-			return (
-				<CompactCard
-					key={ `email-row-${ email.mailbox }` }
-					className="email-plan-mailboxes-list__mailbox-list-item"
-				>
-					<MaterialIcon icon="email" />
-					<span>
-						{ email.mailbox }@{ email.domain }
-					</span>
-					{ isEmailUserAdmin( email ) && (
-						<Badge type="info">
-							{ translate( 'Admin', {
-								comment: 'Email user role displayed as a badge',
-							} ) }
-						</Badge>
-					) }
-				</CompactCard>
-			);
-		} );
+function EmailPlanMailboxesList( { emails, isLoadingEmails } ) {
+	const translate = useTranslate();
 
+	const MailboxListHeader = ( { children } ) => {
 		return (
 			<div className="email-plan-mailboxes-list__mailbox-list">
 				<SectionHeader label={ translate( 'Mailboxes' ) } />
-				{ emailsItems }
+				{ children }
 			</div>
 		);
+	};
+
+	const MailboxListItem = ( { children, isPlaceholder, noEmails } ) => {
+		const className = classNames( 'email-plan-mailboxes-list__mailbox-list-item', {
+			'is-placeholder': isPlaceholder,
+			'no-emails': noEmails,
+		} );
+		return <CompactCard className={ className }>{ children }</CompactCard>;
+	};
+
+	if ( isLoadingEmails ) {
+		return (
+			<MailboxListHeader>
+				<MailboxListItem isPlaceholder>
+					<span> </span>
```

Comment:
Wouldn't the following be simpler?

```suggestion
					<span />
```